tests/unit/kvs/kvs_cursor_test.c

```diff
@@ -93,100 +93,97 @@ insert_key_multiple(struct mtf_test_info *lcl_ti, char *pfx, uint nkeys)
 }
 
 void
-create_cursor(
-    struct mtf_test_info *  lcl_ti,
-    char *                  pfx,
-    uint                    nkeys,
-    bool                    reverse,
-    struct hse_kvs_cursor **cur_out)
+expect_key(struct mtf_test_info *lcl_ti, const void *key, size_t klen, const char *expected)
 {
-    struct hse_kvs_cursor *cur;
-    merr_t                 err;
-
-    cur = kvs_cursor_alloc(kvs, pfx, strlen(pfx), reverse);
-    ASSERT_NE(NULL, cur);
+    int rc;
 
-    err = kvs_cursor_init(cur, NULL);
-    ASSERT_EQ(0, err);
+    ASSERT_TRUE(strlen(expected) == klen);
 
-    *cur_out = cur;
+    rc = strncmp(key, expected, klen);
+    ASSERT_EQ(0, rc);
 }
 
 void
-expect_key(struct mtf_test_info *lcl_ti, struct kvs_ktuple *kt_found, char *expected)
+expect_pfx(struct mtf_test_info *lcl_ti, const void *key, size_t klen, const char *pfx)
 {
     int rc;
 
-    rc = strcmp(kt_found->kt_data, expected);
+    ASSERT_TRUE(strlen(pfx) <= klen);
+
+    rc = strncmp(pfx, key, strlen(pfx));
```

Comment:
Maybe use strnlen since you know the max size of a prefix from hse/limits.h

Reply:
I don't see value in `strnlen` when the input string is properly terminated.